cluster-autoscaler/cloudprovider/clusterapi/clusterapi_nodegroup_test.go

```diff
@@ -1361,39 +1340,29 @@ func TestNodeGroupTemplateNodeInfo(t *testing.T) {
 				expectedErr:  nil,
 				nodeLabels: map[string]string{
 					"kubernetes.io/os":                 "windows",
+					"beta.kubernetes.io/os":            "windows",
 					"kubernetes.io/arch":               "arm64",
+					"beta.kubernetes.io/arch":          "arm64",
 					"node.kubernetes.io/instance-type": "instance1",
-					"anotherLabel":                     "nodeValue", // This should not be copied as it is not a well known label
-				},
-				nodegroupLabels: map[string]string{
```

Comment:
is it the case that nodegroup labels will be added to node(s) under the `.metadata/labels` level?

Reply:
yes, when the node is created the labels will be added. but, we currently don't have a mechanism from the cluster-api side about where to find those labels, this issue is related https://github.com/kubernetes-sigs/cluster-api/issues/7006